src/App.tsx

```diff
@@ -8,37 +8,194 @@ import { PostsList } from './components/PostsList';
 import { PostDetails } from './components/PostDetails';
 import { UserSelector } from './components/UserSelector';
 import { Loader } from './components/Loader';
+import { Post } from './types/Post';
+import { User } from './types/User';
+import { Comment } from './types/Comment';
+import {
+  getPosts,
+  getUsers,
+  getComments,
+  postComment,
+  deleteComment,
+} from './api/post';
 
 export const App: React.FC = () => {
+  const [posts, setPosts] = useState<Post[]>([]);
+  const [users, setUsers] = useState<User[]>([]);
+  const [selectedUserName, setSelectedUserName] = useState('');
+  const [selectedUser, setSelectedUser] = useState<number | null>(null);
+  const [isUsersLoading, setIsUsersLoading] = useState(false);
+  const [isPostsLoading, setIsPostsLoading] = useState(false);
+  const [isNoPosts, setIsNoPosts] = useState(false);
+  const [selectedPost, setSelectedPost] = useState<Post | null>(null);
+  const [loadComments, setLoadComments] = useState<Comment[]>([]);
+  const [isCommentsLoad, setIsCommentsLoad] = useState(false);
+  const [isNoComments, setIsNoComments] = useState(false);
+  const [isShowForm, setIsShowForm] = useState(false);
+  const [isShowButton, setIsShowButton] = useState(false);
+  const [isCommentsUpdate, setIsCommentsUpdate] = useState(false);
+  const [isNewCommentLoad, setIsNewCommentLoad] = useState(false);
+  const [isCommentDelete, setIsCommentDelete] = useState(false);
+
+  useEffect(() => {
+    const loadUsers = async () => {
+      try {
+        const usersList = await getUsers();
+
+        setUsers(usersList);
+      } catch {
+        setIsUsersLoading(true);
```

Comment:
```suggestion
        setIsUsersLoadingError(true);
```

better to specify that is error